### PR TITLE
Replaced 'kubeedge/pause' with 'k8s.gcr.io/pause' multi arch image

### PR DESCRIPTION
**Pull request by siredmar**
What type of PR is this?
/kind feature

What this PR does / why we need it:
This PR is a spinoff of #3098 with only the needed changes

This PR uses official pause container image which are already multiarch images. Therefore the need to explicitly define the arch for the pause image wonn't be needed anymore. Examples kubeedge/pause, kubeedge/pause-arm64 and kubeedge/pause-arm.

Does this PR introduce a user-facing change?:

NONE

### CODE REVIEW

**Comment by siredmar** [comment]
This is blocked until there is a sync between the multiarch image `k8s.gcr.io/pause` images and `kubeedge/pause`.

File: common/constants/default.go [hunk 1]
```diff
@@ -45,8 +45,6 @@ const (
 	DefaultRemoteRuntimeEndpoint       = "unix:///var/run/dockershim.sock"
 	DefaultRemoteImageEndpoint         = "unix:///var/run/dockershim.sock"
 	DefaultPodSandboxImage             = "kubeedge/pause:3.1"
```
Comment (chendave) [inline common/constants/default.go:47]:
```
# uname -i
aarch64

# docker image inspect kubeedge/pause:3.1 | grep "Architecture"
        "Architecture": "amd64",
```

we need to push the multi-arch docker image to docker hub first. @fisherxu  who has the permission to update this docker image?
https://hub.docker.com/r/kubeedge/pause/tags?page=1&ordering=last_updated

**Comment by chendave** [comment]
/hold
until the upstream docker image is synced and updated.

**Comment by fisherxu** [comment]
@chendave @siredmar @subpathdev I have created the multi-arch docker image `kubeedge/pause:3.1` to the docker hub, would you like to have a look and check it work expected? Thanks! 

Ref: https://hub.docker.com/r/kubeedge/pause/tags?page=1

**Review by fisherxu — COMMENTED** [review]
/hold cancel
/approve

**Review by fisherxu — COMMENTED** [review]
/lgtm

**Pull request merged**